### PR TITLE
Update code for version 1.2 of metadata schema

### PR DESCRIPTION
**Pull request by phynes-sensiblecode**

### CODE REVIEW

File: test/test_integration.py [hunk 1]
```diff
@@ -83,19 +83,22 @@ def test_generated_json(self, mock_date):
...
+            self.assertEqual(dataset_metadata, expected_dataset_metadata,
+                             msg=f'Comparing out/{FILENAME_DATASET} and expected/dataset-metadata.json')
 
             with open(os.path.join(output_dir, FILENAME_TABLES)) as f:
                 table_metadata = json.load(f)
             with open(os.path.join(file_dir, 'expected/table-metadata.json')) as f:
                 expected_table_metadata = json.load(f)
-            self.assertEqual(table_metadata, expected_table_metadata)
+            self.assertEqual(table_metadata, expected_table_metadata,
+                             f'Comparing out/{FILENAME_TABLES} and expected/dataset-metadata.json')
```
Comment (phynes-sensiblecode) [inline test/test_integration.py:101]: Should be `expected/table-metadata.json`

File: ctb_metadata_files/metadata.graphql [hunk 2]
```diff
@@ -106,4 +107,14 @@ type Statistical_Unit {
   Statistical_Unit: String!
   Statistical_Unit_Description: String!
 }
- 
+
+type Observation_Type {
```
[the remark this answers is not in the document]
Comment (SpencerHedger) [reply ctb_metadata_files/metadata.graphql:112]: Discussed with @phynes-sensiblecode and this is something that can be changed later when the aforementioned PR is completed.

**Review by SpencerHedger — CHANGES_REQUESTED** [review]
Looks good, my comment on observation types has been discussed already so that leaves just the error you subsequently spotted and commented on (https://github.com/sensiblecodeio/csv-to-cantabular-metadata-2021/pull/21#discussion_r921587046) to be fixed.

**Review by SpencerHedger — APPROVED** [review]
Looks good.

**Pull request merged**